packages/components/src/SegmentedControl/SegmentedControl.module.css

```diff
@@ -0,0 +1,62 @@
+.container {
+  display: grid;
+  grid-template-columns: repeat(var(--segmentedControl--option-count), 1fr);
+  align-items: center;
+  position: relative;
+  min-width: max-content;
+  border-radius: var(--radius-base);
+  border: var(--border-base) solid var(--color-border--interactive);
+  background-color: var(--color-surface--background);
+}
+
+.container input[type="radio"] {
+  position: absolute;
+  left: -999vw;
+  cursor: pointer;
+}
+
+.container label {
+  z-index: 1;
+  padding: var(--space-smaller) 12px;
```

Comment:
@chris-at-jobber I see in figma the padding is `--space-smaller` 
 `12px`; let me know how this feels to you.

Reply:
I think there's something iffy about the lineheight (in Figma vs code) because in Figma, the output is a 32px high element (which matches the height of a `small` Button):

![image](https://github.com/user-attachments/assets/7af3f704-5cfd-4c66-a8bd-d638be0a46d7)

Thinking more about the vertical padding... I'm noticing that even some of the example usages we see SegmentedControl placed in line with both small and base Buttons, so maybe we need to leverage similar sizes? 

Something like `smallest` padding, but a min-height for a `small` and `base` SegmentedControl that lines up with Button? (32px small, 40 base, etc) 

![image](https://github.com/user-attachments/assets/7959a920-53eb-4ca0-8b55-67cb7c889416)
![image](https://github.com/user-attachments/assets/44b6e5e2-73b0-49a9-8365-f3f00f836c5b)
![image](https://github.com/user-attachments/assets/ff4bc6da-1d00-4605-ab3d-13a1bdcbea4f)11

Or if there's something we can do CSS-wise that defaults to SegmentedControl taking up the same height as its' container, but maybe that's too generic and/or prone to breaking 🤔

Reply:
(Configuring the size can be a follow-up if need be, just realizing that we see different sizes in use already)

Reply:
I've added a subtask as a fast follow to add an optional size prop for `base` and `small` - really shouldn't take much of a lift and I can branch off this branch while it's being reviewed 👍🏻